routing/routing_session.hpp

```diff
@@ -35,7 +38,19 @@ struct SpeedCameraRestriction
   SpeedCameraRestriction() : m_index(0), m_maxSpeedKmH(numeric_limits<uint8_t>::max()) {}
 };
 
-class RoutingSession
+class TrafficCache
```

Comment:
Actually, my suggestion was:
```c++
class TrafficInfoGetter
{
public:
  virtual ~TrafficInfoGetter() = default;
  virtual void Set(traffic::TrafficInfo && info);
  virtual shared_ptr<traffic::TrafficInfo> Get(MwmSet::MwmId const & mwmId) const;
  virtual void Remove(MwmSet::MwmId const & mwmId);
  virtual void Clear();

private:
  map<MwmSet::MwmId, shared_ptr<...>> m_cache;
};
```
The default implementation should be the same as TrafficCache, and actually you don't need TrafficCache at all. Also, you don't need TrafficInfoGetterForTesting, as this implementation will do all what you need.